calendar/calendar-tool/tool/src/webapp/js/sakai-calendar.js

```diff
@@ -1,110 +1,155 @@
-  const siteId = window.top.portal.siteId;
-  // We need the updated calendar events, do not cache requests.
-  const requestHeaders = new Headers();
-  requestHeaders.append('pragma', 'no-cache');
-  requestHeaders.append('cache-control', 'no-cache');
-  const requestInit = {
-    method: 'GET',
-    headers: requestHeaders,
-  };
-  const calendarDiv = document.getElementById('calendarDiv');
-  // Initialize fullcalendar and render it in the calendarDiv.
-  const calendar = new FullCalendar.Calendar(calendarDiv, {
-    initialView: 'timeGridWeek',
-    timeZone: userTimeZone,
-    displayEventTime: false,
-    allDaySlot: false,
-    headerToolbar: {
-      left: 'prev,next today',
-      center: 'title',
-      right: 'dayGridMonth,timeGridWeek,timeGridDay,listWeek'
-    },
-    eventSources: [{
-      events: function(event, successCallback, failureCallback) {
-        const requestStartDate = moment(event.start).format('YYYY-MM-DD');
-        const requestEndDate = moment(event.end).format('YYYY-MM-DD');
-        const requestData = {
-          merged: true,
-          firstDate: requestStartDate,
-          lastDate: requestEndDate
-        };
-        // Fetch the site calendar events from the REST endpoint.
-        const calendarRequestUrl = new URL(`${sakaiOrigin}/direct/calendar/site/${siteId}.json`);
-        Object.keys(requestData).forEach(key => calendarRequestUrl.searchParams.append(key, requestData[key]));
-        window.fetch(calendarRequestUrl, requestInit)
-        .then(response => {
-          if (response.ok) {
-            return response.json()
-          } else throw new Error("Network error while fetching calendar data");
-        })
-        .then(responseData => {
-          const events = [];
-          responseData.calendar_collection.forEach(event => {
-            // Every Sakai Calendar event has to be mapped with a full calendar event.
-            const startDate = moment(new Date(event.firstTime.time)).tz(userTimeZone).format();
-            const endDate = moment(new Date(event.firstTime.time + event.duration)).tz(userTimeZone).format();
-            // The calendar event url needs to be a link to view or edit the event.
-            const eventReference = event.eventReference;
-            const eventLink = new URL(window.location);
-            eventLink.searchParams.append("eventReference", eventReference);
-            eventLink.searchParams.append("sakai_action", "doDescription");
-            eventLink.searchParams.append("panel", "Main");
-            events.push({
-              url: eventLink.href,
-              title: event.title,
-              start: startDate,
-              site_name: event.siteName,
-              type: event.type,
-              icon: event.eventIcon,
-              event_id:  event.eventId,
-              attachments: event.attachments,
-              eventReference: event.eventReference,
-              end: endDate
+const SakaiCalendar = {
```

Comment:
I'd go for sakaiCalendar as this is not a class, it's an object, a member variable of window. Seems pedantic, but with ES6 we now have a lot of classes floating about and they may have static members on them.

Reply:
Okay, makes sense.